src/core/toga/fonts.py

```diff
@@ -34,3 +36,95 @@ def bind(self, factory):
+        When invalid values for style, variant or weight are passed, NORMAL will be used instead.
+
+        Platforms that support dynamic font loading will load the font from the supplied path when a toga style
+        references the registered font, e.g.
+
+        Font.register("Font Awesome 5 Free Solid", "resources/Font Awesome 5 Free-Solid-900.otf")
+
+        Font.register("Roboto", "resources/Roboto-Bold.ttf", weight=Font.BOLD)
+
+        Args:
+            family (str):  The font family name
+            path (str):    The path to the font file, relative to the application's module directory.
+            weight (str):  The font weight: Font.NORMAL (default) or a value from Font.FONT_WEIGHTS
+            style (str):   The font style: Font.NORMAL (default) or a value from Font.FONT_STYLES
+            variant (str): The font variant: Font.NORMAL (default) or a value from Font.FONT_VARIANTS
+        """
+        registered_font = RegisteredFont(
+            family, path, weight=weight, style=style, variant=variant
+        )
+        _REGISTERED_FONT_CACHE[registered_font.key] = registered_font
+
+    @staticmethod
+    def find_registered_font(family, weight=NORMAL, style=NORMAL, variant=NORMAL):
+        """
+        Returns the RegisteredFont which has been registered with Font.register()
+        When the exact family/weight/style/variant combination has not been registered, the method tries
+        to return the RegisteredFont for the registered family only, e.g.
+
+        registered_font = Font.find_registered_font("Roboto", weight=Font.BOLD)
+
+        Args:
+            family (str):  The font family name
+            weight (str):  The font weight: Font.NORMAL (default) or a value from Font.FONT_WEIGHTS
+            style (str):   The font style: Font.NORMAL (default) or a value from Font.FONT_STYLES
+            variant (str): The font variant: Font.NORMAL (default) or a value from Font.FONT_VARIANTS
+        Returns:
+            The RegisteredFont or None
+        """
+        font_key = RegisteredFont.make_key(family, weight, style, variant)
+        try:
+            return _REGISTERED_FONT_CACHE[font_key]
+        except KeyError:
+            try:
+                if weight != NORMAL or style != NORMAL or variant != NORMAL:
+                    font_key_family = RegisteredFont.make_key(
+                        family, NORMAL, NORMAL, NORMAL
+                    )
+                    print(
+                        "Registered font "
+                        + font_key
+                        + " not found. Using "
+                        + font_key_family
+                        + " instead."
+                    )
+                    return _REGISTERED_FONT_CACHE[font_key_family]
+                else:
+                    return None
+            except KeyError:
+                print("Registered font " + font_key_family + " not found.")
+                return None
+
+
+class RegisteredFont:
```

Comment:
I'm not sure I see what having a second class gains us here. It's an internal cache lookup; a tuple is more than enough to operate as a key.

Reply:
OK, I addressed this issue